### DISCUSSION TITLE
First time using - Questions/Problems(?)

### DISCUSSION THREAD

**Discussion by jbmorris289** [umami-software/umami#803] [Q&A]
Hello! I hope this is the right place for this kind of post. Can't find where else to post.

So this will be my first time self-hosting anything like this, so I am running into hurdles as I figure this all. I have finally hit a total block.

Following the [instructions](https://umami.is/docs/install) on the site, I just built the app with npm and ran it. Before doing this, I had confusion:

1. What do I do with the HASH_SALT variable? Do I generate something using a tool or it is done on its own?
2. After any change to the .env file, does the program need rebuilding?

Additionally, I had issues with postgresql. May have fixed them, but I received a database authentication error on my first attempt. (My Postgresql user had no password, so I set one. Might have to apply the changes by rebuilding before trying again?)

**Comment by jbmorris289** [comment]
So I did fix the issue with Postgresql. Just needed to change the DB name in the .env file, and actually give my psql username a password.

So now I'm just confused on the HASH_SALT variable.

**Comment by mikecao** [comment]
The HASH_SALT can be any random string. All it does it generate unique values for your installation.

**Comment by jbmorris289** [comment]
> The HASH_SALT can be any random string. All it does it generate unique values for your installation.

So I can just make it anything (might just use random.org)?

**Comment by mufidu** [comment]
Yes, you can.

**Comment by jbmorris289** [comment]
Well, I've got it running fine on my DigitalOcean server, but now I have just one more concern... being indexed by search engines.

Going to need to add a robots.txt file... I'm guessing I place it in the "pages" directory and it won't break anything?

**Comment by mufidu** [comment]
I think you can place it anywhere as long as you can access it via `yourwebsite.com/robots.txt`.

**Comment by jbmorris289** [comment]
Got it to work... I had to add something to the ngnix server block config file. Now umami is running... and I think ready to go.

The only user is admin as stated in the install guide, created when creating the database tables. Besides having to change the password, are there any risks to using this user, or should I create another?

Sorry for the many questions... just trying to make sure I do this right and not fatally screw something up.

**Comment by mufidu** [comment]
It's fine. I personally just use the default admin account. But if you want to be sure, you can add another.

**Comment by flamekiller22** [comment]
Hey I wanted to know how to use the API for umami. I have been able to use the basic features but couldn't understand how to add the parameters like start_at, end_at etc.
I am using postman for learning how to use the API.
![umami api trial](https://user-images.githubusercontent.com/90386026/150587264-59095e45-d914-4397-8c6b-f0bc43a49fb9.jpg)

**Comment by mikecao** [comment]
@flamekiller22 The `start_at` and `end_at` values are in unix time

**Reply by flamekiller22** [reply]
Thanks ^_^

**Comment by t3brightside** [comment]
Maybe it would make sense the HASH_SALT being generated automatically and not to bother the user in install process?
Or offer an option to generate one? I was wondering why I'm being asked for one too...
